### ISSUE TITLE
Synchronous XHR request is giving Chrome console warning

### ISSUE THREAD

**Issue by AndrewGibson27** [NUKnightLab/TimelineJS3#363]
I'm loading a timeline onto a page when a user clicks a button. But Chrome is giving me the following console warning regarding the synchronous nature of XHR request:

`Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.`

After inspecting the code, it seems like it has to with this block of code (lines 2197-2208 in the [CDN](https://cdn.knightlab.com/libs/timeline3/latest/js/timeline.js)):

``` javascript
var async = 'async' in settings ? settings.async : true
xhr.open(settings.type, settings.url, async, settings.username, settings.password)

for (name in headers) nativeSetHeader.apply(xhr, headers[name])

if (settings.timeout > 0) abortTimeout = setTimeout(function(){
    xhr.onreadystatechange = empty
    xhr.abort()
    ajaxError(null, 'timeout', xhr, settings, deferred)
 }, settings.timeout)

// avoid sending empty string (#319)
xhr.send(settings.data ? settings.data : null)
```

Here's what my code looks like (the CDN is included on my page):

``` javascript
document.getElementById('button').addEventListener('click', function(){
    var timeline = new TL.Timeline('timeline', 'https://docs.google.com/spreadsheets/d/1cWqQBZCkX9GpzFtxCWHoqFXCHg-ylTVUWlnrdYMzKUI/pubhtml');
};
```

**Comment by GuillaumeMasselot** [comment]
Hello,
Could you give an approx. date of resolution or is it a "when it's done" issue ?
Cheers

**Comment by JoeGermuska** [comment]
this is fixed in 8d8591b9a1223f2156f2917aad4dd57fc1c93611 which will be included in the next release

**Comment by JoeGermuska** [comment]
TimelineJS 3.8.12 is released, including this fix.
